### PR TITLE
fix: cannot reference variables in DEFINE statement

### PR DESCRIPTION
**Pull request by spena**
### Description 
Issue: Variables cannot be referenced in the DEFINE statement. i.e.
```
ksql> define env = 'qa';
ksql> define topicName = 'topic_${env}';
ksql> show variables;

 Variable Name | Value    
--------------------------
 env           | qa       
 topicName     | topic_${env}
--------------------------
```

This PR fixes the above issue by looking at the `VariableValueContext` in the `VariableSubstitutor`, and replace any variables referenced there.

Example:
```
ksql> define env = 'qa';
ksql> define topicName = 'topic_${env}';
ksql> show variables;

 Variable Name | Value    
--------------------------
 env           | qa       
 topicName     | topic_qa 
--------------------------
```

### Testing done 
Added unit tests
Verified manually

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/variable-substitution.md [hunk 1]
```diff
@@ -86,6 +86,16 @@ ksqlDB doesn't add single-quotes to values during variable substitution. Also, y
 !!! note
     Variables are case-insensitive. A reference to `${replicas}` is the same as `${REPLICAS}`.
 
+Escape substitution variables
+-----------------------------
+
+If you want to escape a variable reference, so it is not substituted, then use double `$$` characters.
```
Comment (JimGalasyn) [inline docs/developer-guide/variable-substitution.md:92]:
```suggestion
To escape a variable reference, so it isn't substituted, use double dollar-sign characters, `$$`. The following example shows how to escape a variable named `format`.
```

File: docs/developer-guide/variable-substitution.md [hunk 2]
```diff
@@ -86,6 +86,16 @@ ksqlDB doesn't add single-quotes to values during variable substitution. Also, y
...
+```
+DEFINE format = 'AVRO';
+SELECT '$${format}' FROM stream;
+```
+The above query will become `SELECT '${format}' FROM stream`
```
Comment (JimGalasyn) [inline docs/developer-guide/variable-substitution.md:97]:
```suggestion
ksqlDB converts the previous query to `SELECT '${format}' FROM stream`.
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a couple of suggestions.

**Review by agavra — APPROVED** [review]
LGTM!

**Pull request merged**